contracts/zero-ex/contracts/src/features/MultiplexFeature.sol

```diff
@@ -0,0 +1,799 @@
+        // provider.
+
+        // The `outputTokenAmount` returned by `_batchFill` may not
+        // be fully accurate (e.g. due to some janky token and/or
+        // reentrancy situation).
+        outputTokenAmount = fillData.outputToken.getTokenBalanceOf(msg.sender)
+            .safeSub(senderBalanceBefore);
+        require(
+            outputTokenAmount >= minBuyAmount,
+            "MultiplexFeature::batchFill/UNDERBOUGHT"
+        );
+
+        uint256 ethBalanceAfter = address(this).balance;
+        require(
+            ethBalanceAfter >= ethBalanceBefore,
+            "MultiplexFeature::batchFill/OVERSPENT_ETH"
+        );
+        // Refund ETH
+        if (ethBalanceAfter > ethBalanceBefore) {
+            payable(msg.sender).transfer(ethBalanceAfter - ethBalanceBefore);
+        }
+    }
+
+    /// @dev Executes a sequence of fills "hopping" through the
+    ///      path of tokens given by `fillData.tokens`. Refer to the
+    ///      internal variant `_multiHopFill` for the allowed nested
+    ///      operations.
+    /// @param fillData Encodes the path of tokens, the sell amount,
+    ///        and the nested operations for this multi-hop fill.
+    /// @param minBuyAmount The minimum amount of the output token
+    ///        to buy. Reverts if this amount is not met.
+    /// @return outputTokenAmount The amount of the output token bought.
+    function multiHopFill(
+        MultiHopFillData memory fillData,
+        uint256 minBuyAmount
+    )
+        public
+        payable
+        override
+        returns (uint256 outputTokenAmount)
+    {
+        require(
+            fillData.tokens.length == fillData.calls.length + 1,
+            "MultiplexFeature::multiHopFill/MISMATCHED_ARRAY_LENGTHS"
+        );
+        IERC20TokenV06 outputToken = IERC20TokenV06(fillData.tokens[fillData.tokens.length - 1]);
+        uint256 senderBalanceBefore = outputToken.getTokenBalanceOf(msg.sender);
+        uint256 ethBalanceBefore = address(this).balance.safeSub(msg.value);
+
+        _multiHopFill(fillData, msg.value);
+
+        // RFC: I think these balance checks are arguably unnecessary,
+        // see reasoning above.
+
+        // The `outputTokenAmount` returned by `_multiHopFill` may not
+        // be fully accurate (e.g. due to some janky token and/or
+        // reentrancy situation).
+        outputTokenAmount = outputToken.getTokenBalanceOf(msg.sender)
+            .safeSub(senderBalanceBefore);
+        require(
+            outputTokenAmount >= minBuyAmount,
+            "MultiplexFeature::multiHopFill/UNDERBOUGHT"
+        );
+
+        uint256 ethBalanceAfter = address(this).balance;
+        require(
+            ethBalanceAfter >= ethBalanceBefore,
+            "MultiplexFeature::multiHopFill/OVERSPENT_ETH"
+        );
+        // Refund ETH
+        if (ethBalanceAfter > ethBalanceBefore) {
+            payable(msg.sender).transfer(ethBalanceAfter - ethBalanceBefore);
```

Comment:
Same question ~ any difference now between `transfer` and `call{value: x}("");`?

Reply:
yeah the call syntax is good practice 👍